dolphinscheduler-task-plugin/dolphinscheduler-task-flink/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/FlinkDeployMode.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.task.flink;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Flink deploy mode
+ */
+public enum FlinkDeployMode {
```

Comment:
Would you please kindly add ```yarn-session``` and ```yarn-per-job```? Thanks.

Reply:
You are right. Maybe we can refer to the implementation of spark node for spark 2 and spark 3. But this change may be larger. Are you willing to modify it in this PR? It's OK if not. ^_^ @paul8263

Reply:
Hi @SbloodyS ,
Sorry that I did not quite catch what you mean. Should I implement two kinds of command lines for legacy Flink and the latest one respectively?

Reply:
Yes. Just like the way we do in spark task. @paul8263 
![](https://vip1.loli.io/2022/04/21/ejUfRMbv6N2TcSK.png)